src/components/Todos/TodosList.js

```diff
@@ -0,0 +1,61 @@
+import axios from "axios";
+import React, { useEffect, useState, useContext } from "react";
+import { useHistory } from "react-router";
+import { NavLink } from "react-router-dom";
+import Card from "../UI/Card/Card";
+import classes from "./TodosList.module.css";
+import AuthContext from "../../store/AuthContext";
+
+const TodosList = () => {
+    const [todos, setTodos] = useState(null);
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(null);
+    const authCtx = useContext(AuthContext);
+    const history = useHistory();
+    const token = authCtx.token;
+
+    useEffect(() => {
+        setIsLoading(true);
+        axios.get('http://localhost:5151/todos', {
+            headers: {
+                "Authorization": token
+            }
+        })
+            .then(response => {
+                setTodos(response.data.data);
+                setError(null);
+                console.log(response.data.data);
+            })
+            .catch((error) => {
+                if (error.response) {
+                    if (error.response.status == 401) {
+                        history.push("/login");
+                        return;
+                    }
+                }
+                setError(error.data);
+            });
+        setIsLoading(false);
```

Comment:
1. Very important:: please check the comment at ticket https://trello.com/c/Q6VQuV5Q/257-reacthomework3#comment-60dbff12a9b3703448161e57 . Note that if you `  setIsLoading(false);` here, it will be execute before axios response.  Because it's very important, please read the comment and links in comment carefully.
2. After you understand the issue, use below syntax of axios to handle loading status:
```js
axios.getOrPost('/v1/api_endpoint?with_parameters')
  .then((res) => { // handle success })
  .catch((err) => { // handle error })
  .then(() => { // always executed, you should hide loading here })  
```